packages/Operators/src/PointCloud/DTK_NodeToNodeOperator_impl.hpp

```diff
@@ -0,0 +1,252 @@
+	target_support_ids[entity_counter] = target_node_supports[0];
+	range_space->localMap()->centroid(
+	    *range_entity, target_centers(DIM*entity_counter,DIM) );
+    }
+
+    // Calculate an approximate neighborhood distance for the local target
+    // centers. 
+    double target_proximity = 0.0;
+
+    // Get the local bounding box.
+    Teuchos::Tuple<double,6> local_box;
+    range_space->entitySet()->localBoundingBox( local_box );
+
+    // Calculate the largest span of the cardinal directions.
+    target_proximity = local_box[3] - local_box[0];
+    for ( int d = 1; d < DIM; ++d )
+    {
+        target_proximity = std::max( target_proximity,
+                                     local_box[d+3] - local_box[d] );
+    }
+
+    // Take the proximity to be 0.001 of the largest distance.
+    target_proximity *= 1.0e-3;
+    
+    // Gather the source centers that are in the proximity of the target
+    // centers on this proc.
+    Teuchos::Array<double> dist_sources;
+    CenterDistributor<DIM> distributor( 
+	comm, source_centers(), target_centers(), target_proximity, dist_sources );
+
+    // Gather the global ids of the source centers that are within the proximity of
+    // the target centers on this proc.
+    Teuchos::Array<GO> dist_source_support_ids( distributor.getNumImports() );
+    Teuchos::ArrayView<const GO> source_support_ids_view = source_support_ids();
+    distributor.distribute( source_support_ids_view, dist_source_support_ids() );
+
+    // Build the source/target pairings by finding the nearest neighbor - this
+    // should be the exact same node.
+    SplineInterpolationPairing<DIM> pairings( 
+	dist_sources, target_centers(), true, 1, 0.0 );
+   
+    // Build the coupling matrix.
+    d_coupling_matrix = Teuchos::rcp(
+        new Tpetra::CrsMatrix<Scalar,LO,GO>(range_map, 1) );
+    Teuchos::Array<GO> indices( 1 );
+    Teuchos::Array<double> values( 1, 1.0 );
+    int nn = 0;
+    for ( int i = 0; i < local_num_tgt; ++i )
+    {
+	// If there is no support for this target center then do not build a
+	// local basis.
+	if ( 0 < pairings.childCenterIds(i).size() )
+	{
+            // If we have a neighbor then there should be only 1.
+            DTK_CHECK( 1 == pairings.childCenterIds(i).size() );
+            
+            // Check that our neighbor node has the same coordinates.
+            DTK_CHECK(
+                std::abs(
+                    EuclideanDistance<DIM>::distance(
+                        dist_sources(DIM*pairings.childCenterIds(i)[0],DIM).getRawPtr(),
+                        target_centers(DIM*i,DIM).getRawPtr()) )
+                < 1.0e-14 );
+                
+            // Get the id of the domain node
+            indices[0] =
+                dist_source_support_ids[ pairings.childCenterIds(i)[0] ];
+            
+	    // Populate the coupling matrix row.            
+	    d_coupling_matrix->insertGlobalValues( 
+		target_support_ids[i], indices(), values() );
+	}
```

Comment:
Right - in Srikanth's case he was originally doing a node-to-node map with overlap so some points didn't get data.